src/helpers/utils.ts

```diff
@@ -45,8 +45,25 @@ export async function getEmailAddresses(email: string) {
   ]);
 }
 
-export async function getVerifiedSubscriptions() {
-  return await db.queryAsync('SELECT email, address FROM subscribers WHERE verified > 0');
+export async function getVerifiedSubscriptions(batchSize = 1000) {
+  let page = 0;
+  let results: SqlRow[] = [];
+
+  while (true) {
+    const result = await db.queryAsync(
+      'SELECT email, address FROM subscribers WHERE verified > 0 ORDER BY created LIMIT ? OFFSET ?',
+      [batchSize, page * batchSize]
+    );
+
+    if (result.length === 0) {
+      break;
+    }
+
+    page += 1;
+    results = results.concat(result);
+  }
+
+  return results;
```

Comment:
the problem with this approach is that instead of making a huge request to DB we will exhaust RAM (storing a huge array in memory). I think the solution for this is to take a small batch of email addresses, process them, take another, and so on and so forth.
You can consider converting `getSubscribersEmailFollowingSpace` to an iterator - just an idea, maybe you can come up with a better one.

Reply:
I don't think that this is overkill. But it depends on the number of users and amount of activities in prod. In the worst-case scenario, our server will lay down and messages will not be sent.

@aurelianoB can we merge MR with this?

Reply:
I tend to agree with @wa0x6e, we should go in stages